### ISSUE TITLE
Admin's creds not working with backend on Postgresql

### ISSUE THREAD

**Issue by Sarevok933219** [h44z/wg-portal#202]
Hello. I'm using and testing V2 version. Trying to config with Postgres DB as backend. My config.yaml:
```
advanced:
  log_level: trace
  log_pretty: true
  use_ip_v6: false
core:
  listeningAddress: :8123
  externalUrl: http://localhost:8123
  admin_user: admin@admin.mail.ru
  admin_password: <supersecretkey>
  config_storage_path: /etc/wireguard
  import_existing: true

database:
  type: postgres
  dsn: host=postgres user=<admin> password=<superkey>a dbname=wireguard port=5432 sslmode=disable TimeZone=<localZone>
  debug: true
web:
  site_company_name: WIREGUARD VPN
  external_url: http://localhost:8123
  listening_address: :8123
  request_logging: true

auth:
  callback_url_prefix: http://localhost:8123/api/v0
```
Container is running up. Connection with PSQL is ok. I see tables and some information in them. My supers user is also present in dp. But then I try to start UI I get an errors:
![WG_ERR](https://github.com/h44z/wg-portal/assets/56883129/0dcd2e65-885d-4506-a0eb-7c624678c54b)
...and can't login with my creds.
![WG_ERR2](https://github.com/h44z/wg-portal/assets/56883129/025a125e-4359-48c9-aad5-67c3e64ffc40)

 Where may be mistake? Please help.

**Comment by Sarevok93219** [comment]
The Errors Backend Connection Failure are gone (change config file: `external_url: http://my_ext_ip:8123`), but still can't be authorized by my admin's creds.

**Comment by viktor-zharikov** [comment]
i have that problem, and found some solve, if you use
`  externalUrl: http://localhost:8123` in config, you must use http://localhost:8123 in browser, NOT 127.0.0.1:8123

**Comment by Sarevok93219** [comment]
> i have that problem, and found some solve, if you use ` externalUrl: http://localhost:8123` in config, you must use http://localhost:8123 in browser, NOT 127.0.0.1:8123

Thx. I use http://127.0.0.1:8123 in my yml file. I tried to use localhost but in didn't help. I still can't log in. Please if You can, give me config file and what is under key auth. I don't quite understand what should be here.
```
auth:
  callback_url_prefix: http://127.0.0.1:8123/api/v0

```

**Comment by viktor-zharikov** [comment]
> > i have that problem, and found some solve, if you use ` externalUrl: http://localhost:8123` in config, you must use http://localhost:8123 in browser, NOT 127.0.0.1:8123
> 
> Thx. I use http://127.0.0.1:8123 in my yml file. I tried to use http://localhost:8123 instead, but it didn't help. I still can't log in. Please if You can, give me config file and what is under key auth. I don't quite understand what should be here.
> 
> ```
> auth:
>   callback_url_prefix: http://127.0.0.1:8123/api/v0
> ```
> 
> my config file looks like:
> 
> ```
> advanced:
>   log_level: trace
>   log_pretty: true
>   log_json: true
>   use_ip_v6: false
>   config_storage_path: /etc/wireguard
> 
> core:
>   admin_user: admin@mail.local.ru
>   admin_password: <supersecretkey>
>   import_existing: true
>   self_provisioning_allowed: true
> 
> database:
>   #type: sqlite
>   #dsn: /app/data/keys.db
>   # debug: true
>   type: postgres
>   dsn: host=postgres user=admin password=<admin's pass> dbname=wireguard port=5432 sslmode=disable TimeZone=Europe/Moscow
>   debug: true
> 
> web:
>   site_company_name: WIREGUARD VPN MYCOMPANYNAME
>   external_url: http://localhost:8123
>   #external_url: http://127.0.0.1:8123
>   listening_address: :8123
>   request_logging: true
> 
> auth:
>   callback_url_prefix: http://localhost:8123/api/v0
>   #callback_url_prefix: http://127.0.0.1:8123/api/v0
>   #callback_url_prefix: /api/v0
> ```

In you example config 'external_url: http://localhost:8123' and 'callback_url_prefix: http://127.0.0.1:8123/api/v0' is different, make all url in one style, at first try to localhost, and use "Ctrl + Shift + I opens the developer tools for the browser" network tab

**Comment by Sarevok93219** [comment]
> > > i have that problem, and found some solve, if you use ` externalUrl: http://localhost:8123` in config, you must use http://localhost:8123 in browser, NOT 127.0.0.1:8123
> > 
> > 
> > Thx. I use http://127.0.0.1:8123 in my yml file. I tried to use http://localhost:8123 instead, but it didn't help. I still can't log in. Please if You can, give me config file and what is under key auth. I don't quite understand what should be here.
> > ```
> > auth:
> >   callback_url_prefix: http://127.0.0.1:8123/api/v0
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > my config file looks like:
> > ```
> > advanced:
> >   log_level: trace
> >   log_pretty: true
> >   log_json: true
> >   use_ip_v6: false
> >   config_storage_path: /etc/wireguard
> > 
> > core:
> >   admin_user: admin@mail.local.ru
> >   admin_password: <supersecretkey>
> >   import_existing: true
> >   self_provisioning_allowed: true
> > 
> > database:
> >   #type: sqlite
> >   #dsn: /app/data/keys.db
> >   # debug: true
> >   type: postgres
> >   dsn: host=postgres user=admin password=<admin's pass> dbname=wireguard port=5432 sslmode=disable TimeZone=Europe/Moscow
> >   debug: true
> > 
> > web:
> >   site_company_name: WIREGUARD VPN MYCOMPANYNAME
> >   external_url: http://localhost:8123
> >   #external_url: http://127.0.0.1:8123
> >   listening_address: :8123
> >   request_logging: true
> > 
> > auth:
> >   callback_url_prefix: http://localhost:8123/api/v0
> >   #callback_url_prefix: http://127.0.0.1:8123/api/v0
> >   #callback_url_prefix: /api/v0
> > ```
> 
> In you example config 'external_url: http://localhost:8123' and 'callback_url_prefix: http://127.0.0.1:8123/api/v0' is different, make all url in one style, at first try to localhost, and use "Ctrl + Shift + I opens the developer tools for the browser" network tab

I'm not sure, but may be something wrong when interacting with backend (postgres DB)
![image](https://github.com/h44z/wg-portal/assets/133360668/e3f63285-b314-4d53-babe-19625b517214)

**Comment by viktor-zharikov** [comment]
> > > > i have that problem, and found some solve, if you use ` externalUrl: http://localhost:8123` in config, you must use http://localhost:8123 in browser, NOT 127.0.0.1:8123
> > > 
> > > 
> > > Thx. I use http://127.0.0.1:8123 in my yml file. I tried to use http://localhost:8123 instead, but it didn't help. I still can't log in. Please if You can, give me config file and what is under key auth. I don't quite understand what should be here.
> > > ```
> > > auth:
> > >   callback_url_prefix: http://127.0.0.1:8123/api/v0
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > my config file looks like:
> > > ```
> > > advanced:
> > >   log_level: trace
> > >   log_pretty: true
> > >   log_json: true
> > >   use_ip_v6: false
> > >   config_storage_path: /etc/wireguard
> > > 
> > > core:
> > >   admin_user: admin@mail.local.ru
> > >   admin_password: <supersecretkey>
> > >   import_existing: true
> > >   self_provisioning_allowed: true
> > > 
> > > database:
> > >   #type: sqlite
> > >   #dsn: /app/data/keys.db
> > >   # debug: true
> > >   type: postgres
> > >   dsn: host=postgres user=admin password=<admin's pass> dbname=wireguard port=5432 sslmode=disable TimeZone=Europe/Moscow
> > >   debug: true
> > > 
> > > web:
> > >   site_company_name: WIREGUARD VPN MYCOMPANYNAME
> > >   external_url: http://localhost:8123
> > >   #external_url: http://127.0.0.1:8123
> > >   listening_address: :8123
> > >   request_logging: true
> > > 
> > > auth:
> > >   callback_url_prefix: http://localhost:8123/api/v0
> > >   #callback_url_prefix: http://127.0.0.1:8123/api/v0
> > >   #callback_url_prefix: /api/v0
> > > ```
> > 
> > 
> > In you example config 'external_url: http://localhost:8123' and 'callback_url_prefix: http://127.0.0.1:8123/api/v0' is different, make all url in one style, at first try to localhost, and use "Ctrl + Shift + I opens the developer tools for the browser" network tab
> 
> I'm not sure, but may be something wrong when interacting with backend (postgres DB) ![image](https://private-user-images.githubusercontent.com/133360668/294276879-e3f63285-b314-4d53-babe-19625b517214.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ0MjEwODIsIm5iZiI6MTcwNDQyMDc4MiwicGF0aCI6Ii8xMzMzNjA2NjgvMjk0Mjc2ODc5LWUzZjYzMjg1LWIzMTQtNGQ1My1iYWJlLTE5NjI1YjUxNzIxNC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMTA1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDEwNVQwMjEzMDJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05Yjk1ZmE4Y2FhNGRhZDdjZjM5NjIxZWExNDFiNmFiOWY0MGZiYjJkYjgxODA2MmEyZjUwN2M5MTljN2VjMTMyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.BVAYWZYCrvVBzTVw2qtOv1w5mhS2aM11v50Snqe094M)

Try to up wg with default config file, db, credentials, i am sure it's work

**Comment by Sarevok93219** [comment]
> > > > > i have that problem, and found some solve, if you use ` externalUrl: http://localhost:8123` in config, you must use http://localhost:8123 in browser, NOT 127.0.0.1:8123
> > > > 
> > > > 
> > > > Thx. I use http://127.0.0.1:8123 in my yml file. I tried to use http://localhost:8123 instead, but it didn't help. I still can't log in. Please if You can, give me config file and what is under key auth. I don't quite understand what should be here.
> > > > ```
> > > > auth:
> > > >   callback_url_prefix: http://127.0.0.1:8123/api/v0
> > > > ```
> > > > 
> > > > 
> > > >     
> > > >       
> > > >     
> > > > 
> > > >       
> > > >     
> > > > 
> > > >     
> > > >   
> > > > my config file looks like:
> > > > ```
> > > > advanced:
> > > >   log_level: trace
> > > >   log_pretty: true
> > > >   log_json: true
> > > >   use_ip_v6: false
> > > >   config_storage_path: /etc/wireguard
> > > > 
> > > > core:
> > > >   admin_user: admin@mail.local.ru
> > > >   admin_password: <supersecretkey>
> > > >   import_existing: true
> > > >   self_provisioning_allowed: true
> > > > 
> > > > database:
> > > >   #type: sqlite
> > > >   #dsn: /app/data/keys.db
> > > >   # debug: true
> > > >   type: postgres
> > > >   dsn: host=postgres user=admin password=<admin's pass> dbname=wireguard port=5432 sslmode=disable TimeZone=Europe/Moscow
> > > >   debug: true
> > > > 
> > > > web:
> > > >   site_company_name: WIREGUARD VPN MYCOMPANYNAME
> > > >   external_url: http://localhost:8123
> > > >   #external_url: http://127.0.0.1:8123
> > > >   listening_address: :8123
> > > >   request_logging: true
> > > > 
> > > > auth:
> > > >   callback_url_prefix: http://localhost:8123/api/v0
> > > >   #callback_url_prefix: http://127.0.0.1:8123/api/v0
> > > >   #callback_url_prefix: /api/v0
> > > > ```
> > > 
> > > 
> > > In you example config 'external_url: http://localhost:8123' and 'callback_url_prefix: http://127.0.0.1:8123/api/v0' is different, make all url in one style, at first try to localhost, and use "Ctrl + Shift + I opens the developer tools for the browser" network tab
> > 
> > 
> > I'm not sure, but may be something wrong when interacting with backend (postgres DB) ![image](https://private-user-images.githubusercontent.com/133360668/294276879-e3f63285-b314-4d53-babe-19625b517214.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ0MjEwODIsIm5iZiI6MTcwNDQyMDc4MiwicGF0aCI6Ii8xMzMzNjA2NjgvMjk0Mjc2ODc5LWUzZjYzMjg1LWIzMTQtNGQ1My1iYWJlLTE5NjI1YjUxNzIxNC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMTA1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDEwNVQwMjEzMDJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05Yjk1ZmE4Y2FhNGRhZDdjZjM5NjIxZWExNDFiNmFiOWY0MGZiYjJkYjgxODA2MmEyZjUwN2M5MTljN2VjMTMyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.BVAYWZYCrvVBzTVw2qtOv1w5mhS2aM11v50Snqe094M)
> 
> Try to up wg with default config file, db, credentials, i am sure it's work

The task is using this product with ability to work with postgres. I have already tested  V1. It works fine with MySQL backend. I don't know why v2 has such problem. I saw hee another issue with similar case ( may be testing version contains bugs).  https://github.com/h44z/wg-portal/issues/198 and it is still has status "opened".

**Comment by Sarevok93219** [comment]
> > > > > > i have that problem, and found some solve, if you use ` externalUrl: http://localhost:8123` in config, you must use http://localhost:8123 in browser, NOT 127.0.0.1:8123
> > > > > 
> > > > > 
> > > > > Thx. I use http://127.0.0.1:8123 in my yml file. I tried to use http://localhost:8123 instead, but it didn't help. I still can't log in. Please if You can, give me config file and what is under key auth. I don't quite understand what should be here.
> > > > > ```
> > > > > auth:
> > > > >   callback_url_prefix: http://127.0.0.1:8123/api/v0
> > > > > ```
> > > > > 
> > > > > 
> > > > >     
> > > > >       
> > > > >     
> > > > > 
> > > > >       
> > > > >     
> > > > > 
> > > > >     
> > > > >   
> > > > > my config file looks like:
> > > > > ```
> > > > > advanced:
> > > > >   log_level: trace
> > > > >   log_pretty: true
> > > > >   log_json: true
> > > > >   use_ip_v6: false
> > > > >   config_storage_path: /etc/wireguard
> > > > > 
> > > > > core:
> > > > >   admin_user: admin@mail.local.ru
> > > > >   admin_password: <supersecretkey>
> > > > >   import_existing: true
> > > > >   self_provisioning_allowed: true
> > > > > 
> > > > > database:
> > > > >   #type: sqlite
> > > > >   #dsn: /app/data/keys.db
> > > > >   # debug: true
> > > > >   type: postgres
> > > > >   dsn: host=postgres user=admin password=<admin's pass> dbname=wireguard port=5432 sslmode=disable TimeZone=Europe/Moscow
> > > > >   debug: true
> > > > > 
> > > > > web:
> > > > >   site_company_name: WIREGUARD VPN MYCOMPANYNAME
> > > > >   external_url: http://localhost:8123
> > > > >   #external_url: http://127.0.0.1:8123
> > > > >   listening_address: :8123
> > > > >   request_logging: true
> > > > > 
> > > > > auth:
> > > > >   callback_url_prefix: http://localhost:8123/api/v0
> > > > >   #callback_url_prefix: http://127.0.0.1:8123/api/v0
> > > > >   #callback_url_prefix: /api/v0
> > > > > ```
> > > > 
> > > > 
> > > > In you example config 'external_url: http://localhost:8123' and 'callback_url_prefix: http://127.0.0.1:8123/api/v0' is different, make all url in one style, at first try to localhost, and use "Ctrl + Shift + I opens the developer tools for the browser" network tab
> > > 
> > > 
> > > I'm not sure, but may be something wrong when interacting with backend (postgres DB) ![image](https://private-user-images.githubusercontent.com/133360668/294276879-e3f63285-b314-4d53-babe-19625b517214.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ0MjEwODIsIm5iZiI6MTcwNDQyMDc4MiwicGF0aCI6Ii8xMzMzNjA2NjgvMjk0Mjc2ODc5LWUzZjYzMjg1LWIzMTQtNGQ1My1iYWJlLTE5NjI1YjUxNzIxNC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMTA1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDEwNVQwMjEzMDJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05Yjk1ZmE4Y2FhNGRhZDdjZjM5NjIxZWExNDFiNmFiOWY0MGZiYjJkYjgxODA2MmEyZjUwN2M5MTljN2VjMTMyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.BVAYWZYCrvVBzTVw2qtOv1w5mhS2aM11v50Snqe094M)
> > 
> > 
> > Try to up wg with default config file, db, credentials, i am sure it's work
> 
> The task is using this product with ability to work with postgres. I have already tested V1. It works fine with MySQL backend. I don't know why v2 has such problem. I saw here another issue with similar case (may be testing version contains bugs). #198 and it is still has status "opened".

And You are right, if changing to default db everything is ok!!!

**Comment by h44z** [comment]
should be fixed by #231
